cache/stringcache/string_caches.go

```diff
@@ -11,7 +11,7 @@ import (
 
 type stringCache interface {
 	elementCount() int
-	contains(searchString string) bool
+	contains(searchString string) (bool, string)
```

Comment:
I would not do this with `contains` method. The name implies that the main return type should be boolean. Consider error return value:
```go
contains(searchString string) (bool, err)
```
`err` should be at the end as it is not the main response from method.

Could we rename `contains` to `match` and get rid of boolean?
```go
// returns rule if matched, otherwise empty string
func match(searchString string) (string)
```